cmd/soroban-cli/src/commands/network/container/shared.rs

```diff
@@ -52,60 +126,25 @@ impl fmt::Display for Network {
     }
 }
 
-pub async fn connect_to_docker(docker_host: &Option<String>) -> Result<Docker, Error> {
-    // if no docker_host is provided, use the default docker host:
-    // "unix:///var/run/docker.sock" on unix machines
-    // "npipe:////./pipe/docker_engine" on windows machines
-
-    let host = docker_host
-        .clone()
-        .unwrap_or(DEFAULT_DOCKER_HOST.to_string());
-
-    // this is based on the `connect_with_defaults` method which has not yet been released in the bollard crate
-    // https://github.com/fussybeaver/bollard/blob/0972b1aac0ad5c08798e100319ddd0d2ee010365/src/docker.rs#L660
-    let connection = match host.clone() {
-        // if tcp or http, use connect_with_http_defaults
-        // if unix and host starts with "unix://" use connect_with_unix
-        // if windows and host starts with "npipe://", use connect_with_named_pipe
-        // else default to connect_with_unix
-        h if h.starts_with("tcp://") || h.starts_with("http://") => {
-            Docker::connect_with_http_defaults()
-        }
-        #[cfg(unix)]
-        h if h.starts_with("unix://") => {
-            Docker::connect_with_unix(&h, DEFAULT_TIMEOUT, API_DEFAULT_VERSION)
-        }
-        #[cfg(windows)]
-        h if h.starts_with("npipe://") => {
-            Docker::connect_with_named_pipe(&h, DEFAULT_TIMEOUT, API_DEFAULT_VERSION)
-        }
-        _ => {
-            return Err(Error::UnsupportedURISchemeError {
-                uri: host.to_string(),
-            });
+pub struct Name(Option<String>, Option<Network>);
+impl Name {
+    pub fn new(name: Option<String>, network: Option<Network>) -> Self {
```

Comment:
Perhaps the name check should happen here, the inner value could be one string, and this could return a result.

Reply:
As I pointed out below, Perhaps just a single `String` is enough. Or if you make the inner type public, you can just use `Name(..)`.